src/hotspot/cpu/aarch64/vm_version_aarch64.cpp

```diff
@@ -246,6 +246,19 @@ void VM_Version::initialize() {
     FLAG_SET_DEFAULT(UseCRC32, false);
   }
 
+
+  // Neoverse V1
+  if (_cpu == CPU_ARM && (_model == 0xd40 || _model2 == 0xd40)) {
+    if (FLAG_IS_DEFAULT(UseCryptoPmullForCRC32)) {
+      FLAG_SET_DEFAULT(UseCryptoPmullForCRC32, true);
+    }
+  }
+
+  if (UseCryptoPmullForCRC32 && (!VM_Version::supports_sha3() || !VM_Version::supports_pmull())) {
```

Comment:
Why does `UseCryptoPmullForCRC32` depend on `VM_Version::supports_sha3()`?

Does `VM_Version::supports_pmull()` really check for the `pmull` in the cryptographic extension?

Reply:
> Why does `UseCryptoPmullForCRC32` depend on `VM_Version::supports_sha3()`?

The implementation uses [eor3](https://developer.arm.com/documentation/ddi0602/2022-06/SIMD-FP-Instructions/EOR3--Three-way-Exclusive-OR-) as well, and it is only available when SHA3 is implemented.
 
> Does `VM_Version::supports_pmull()` really check for the `pmull` in the cryptographic extension?

supports_pmull() returns true when CPU_PMULL/HWCAP_PMULL of VM_Version::_features is [enabled](https://github.com/openjdk/jdk/blob/3ba156082b73c4a8e9d890a57a42fb68df2bf98f/src/hotspot/cpu/aarch64/vm_version_aarch64.hpp#L121). HWCAP_PMULL refers to the [functionality](https://docs.kernel.org/arm64/elf_hwcaps.html) implied by ID_AA64ISAR0_EL1.AES == 0b0010, i.e. [PMULL/PMULL2](https://developer.arm.com/documentation/ddi0595/2021-03/AArch64-Registers/ID-AA64ISAR0-EL1--AArch64-Instruction-Set-Attribute-Register-0) instructions operating on 64-bit data quantities.